pkg/controllers/depends_on.go

```diff
@@ -0,0 +1,26 @@
+package controllers
+
+import (
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+)
+
+// isDependsOnJobReachedStatus checks if depends on ReplicatedJob reaches Ready or Complete status.
+func isDependsOnJobReachedStatus(dependsOnJob jobset.DependsOn, dependsOnJobReplicas int32, rJobsStatuses []jobset.ReplicatedJobStatus) bool {
```

Comment:
The function name is a bit confusing to me, maybe we can name it something like `dependencyReachedStatus` or `replicatedJobReachedStatus`?